extras/bindings/fortran/include/precice/preciceFortran.hpp

```diff
@@ -40,6 +43,42 @@ PRECICE_API void precicef_create_(
     int         participantNameLength,
     int         configFileNameLength);
 
+/**
+ * Fortran specifies the communicator as an integer of default kind which is
+ * equivalent to the C type MPI_Fint, but if preCICE is built without MPI then
+ * the MPI header is not available so a different type must be used.
+ */
+#ifndef PRECICE_NO_MPI
+typedef MPI_Fint comm_int_type;
+#else
+typedef int comm_int_type;
```

Comment:
We prefer `using` over `typedef` and use scoped global names:

```suggestion
using PRECICE_COMM_INT_TYPE = MPI_Fint;
#else
using PRECICE_COMM_INT_TYPE = int;
```